code/game/objects/structures/lattice.dm

```diff
@@ -38,8 +38,10 @@
 /obj/structure/lattice/Destroy()
 	var/turf/old_loc = get_turf(src)
 	. = ..()
-	if(old_loc)
+	if(istype(old_loc))
```

Comment:
Minor note: technically `isturf()` is faster than istype, but it doesn't matter here.